src/cli.mjs

```diff
@@ -112,6 +113,50 @@ function buildYargs(argvInput) {
     .command(localCommand)
     .demandCommand()
     .strictCommands(true)
+    .completion(
+      "completion",
+      "Output bash/zsh script to enable shell completions. See command output for installation instructions.",
+    )
+    .completion(
+      "completion",
+      async function (currentWord, argv, defaultCompletions, done) {
+        // this is pretty hard to debug - if you need to, run
+        // `fauna --get-yargs-completions <command> <flag> <string to match>`
+        // for example: `fauna --get-yargs-completions --profile he`
+        // note that you need to have empty quotes to get all matches:
+        // `fauna --get-yargs-completions --profile ""`
+
+        // then, call the done callback with an array of strings for debugging, like:
+        // done(
+        //   [
+        //     `currentWord: ${currentWord}, currentWordFlag: ${currentWordFlag}, argv: ${JSON.stringify(argv)}`,
+        //   ],
+        // );
+        const previousWord = process.argv.slice(-2, -1)[0].replace(/-/g, "");
+        const currentWordFlag = Object.keys(argv)
+          .filter((key) => previousWord === key)
+          .pop();
+
+        // TODO: this doesn't handle aliasing, and it needs to
+        if (
+          currentWord === "--profile" ||
+          currentWordFlag === "profile" ||
+          currentWord === "-p" ||
+          currentWordFlag === "p"
+        ) {
+          done(getProfileCompletions(currentWord, argv));
+        } else if (
+          currentWord === "--database" ||
+          currentWordFlag === "database" ||
+          currentWord === "-d" ||
+          currentWordFlag === "d"
+        ) {
+          done(await getDbCompletions(currentWord, argv));
+        } else {
+          defaultCompletions();
+        }
+      },
+    )
```

Comment:
this is the meat of the change: yargs calls this method when zsh/bash calls `fauna --get-yargs-completions ...`, and then we do some switching based on yargs argv and process.argv and call methods that either sync or async return a list of strings.

we have to use process.argv as well because yargs argv doesn't include the order of the tokens, so we can't easily find `currentWordFlag`...